### PR TITLE
15183 Added feature flags for old and new webchats

### PR DESCRIPTION
**Pull request by JazzarKarim**
*Issue #:* /bcgov/entity#15183

*Description of changes:*
- Added feature flags for the old webchat and the new Genesys web message component.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namerequest license (Apache 2.0).

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -268,6 +270,24 @@ export default class App extends Mixins(
...
     await this.toggleConfirmNrModal(false)
   }
+
+  /** Whether the old webchat should be enabled. */
+  get enableOldWebchat (): boolean {
+    if (getFeatureFlag('enable-web-chat')) {
+      return !!getFeatureFlag('enable-web-chat')
+    } else {
+      return false // Failsafe value
+    }
```
Comment (severinbeauvais) [inline src/App.vue:280]: 1. Please simplify this. Also below.

2. Add the failsafe values to https://github.com/bcgov/namerequest/blob/main/src/plugins/featureFlags.ts

File: src/App.vue [hunk 2]
```diff
@@ -268,6 +270,24 @@ export default class App extends Mixins(
...
+  }
+
+  /** Whether the Genesys web message should be enabled. */
+  get enableGenesysWebMessage (): boolean {
+    if (getFeatureFlag('enable-genesys-web-message')) {
+      return !!getFeatureFlag('enable-genesys-web-message')
+    } else {
+      return true // Failsafe value
```
Comment (severinbeauvais) [inline src/App.vue:288]: Should also fail as False.

**Review by severinbeauvais — COMMENTED** [review]
Logic looks good. Remember the app version :)

File: src/App.vue (hunk 2 above)
Comment (severinbeauvais) [reply src/App.vue]: "No special features if we can't get the FFs from LaunchDarkly -- only base functionality."

Comment (JazzarKarim) [reply src/App.vue]: Yes, makes a lot of sense. Noted.

File: src/App.vue [hunk 3]
```diff
@@ -273,20 +273,12 @@ export default class App extends Mixins(
 
   /** Whether the old webchat should be enabled. */
   get enableOldWebchat (): boolean {
-    if (getFeatureFlag('enable-web-chat')) {
-      return !!getFeatureFlag('enable-web-chat')
-    } else {
-      return false // Failsafe value
-    }
+    return getFeatureFlag('enable-web-chat')
```
Comment (severinbeauvais) [inline src/App.vue:276]: `getFeatureFlag()` returns a type of any. If you use `!!` then you get a boolean instead.

File: src/plugins/featureFlags.ts [hunk 4]
```diff
@@ -12,7 +12,9 @@ const defaultFlagSet: LDFlagSet = {
   'disable-analysis': true, // by default, analysis is disabled
   'hardcoded_regular_wait_time': 0, // by default, use actual wait time
   'hardcoded_priority_wait_time': 0, // by default, use actual wait time
-  'banner-text': '' // by default, there is no banner text
+  'banner-text': '', // by default, there is no banner text
+  'enable-web-chat': false, // by default, old webchat is disabled
+  'enable-genesys-web-message': false // by default, genesys web message is disabled
```
Comment (severinbeauvais) [inline src/plugins/featureFlags.ts:17]: 👍

**Review by severinbeauvais — APPROVED** [review]
LGTM (after any->boolean fix)

File: src/App.vue (hunk 3 above)
Comment (JazzarKarim) [reply src/App.vue:276]: You're right! I was following like this here: https://github.com/bcgov/namerequest/blob/e279a02b010ffaafe8092b69267a070dddb3c8fa/src/components/common/applicant-info-2.vue#L280
I'll fix it.

Comment (severinbeauvais) [reply src/App.vue]: Hah, that other code should be fixed :)

(I think it evaluates the same, but typing is helpful to prevent errors early on.)

Comment (JazzarKarim) [reply src/App.vue]: Got it. Shall I just add the `!!` since I'm already on the subject of feature flags or leave it for another ticket?

Comment (severinbeauvais) [reply src/App.vue]: Sure, fix it.

**Pull request merged**